### ISSUE TITLE
[Feature Request] Run container as supplied UID : GID

### ISSUE THREAD

**Issue by schklom** [benphelps/homepage#488]
### Description

Allow to run without root privileges.

### Other

 ```yaml
services:
  homepage:
    image: ghcr.io/benphelps/homepage
    container_name: homepage
    volumes:
      - ${DOCKER}/homepage/config:/app/config
      - ${DOCKER}/homepage/images:/app/public/images
    ports:
      - 3000:3000
    user: 1001:1001
```
causes warnings such as
```bash
[2022-11-04T21:01:51.627Z] warn: Failed to update prerender cache for /en [Error: EACCES: permission denied, open '/app/.next/server/pages/en.html'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'open',
  path: '/app/.next/server/pages/en.html'
}
```

I think adding
```dockerfile
USER: 1001:1001
```
in the Dockerfile should do the trick. Ideally at the beginning, or near the end if it does not work.

**Comment by JazzFisch** [comment]
This has been talked about to some degree in #95.

**Comment by JigSawFr** [comment]
Works as expected with docker socket proxy :)

**Comment by MrOnak** [comment]
Maybe I got lucky but adding PUID / PGID as environment parameters in docker-compose.yaml works for me. 

```
version: "3.3"
services:
  homepage:
    image: ghcr.io/benphelps/homepage:main
    container_name: homepage
    environment:
      - PUID=1002
      - PGID=100
    networks:
```

**Comment by bartmasz** [comment]
@MrOnak using **environment** instead of **user** in docker-compose.yaml works for me as well. Thanks!

**Comment by schklom** [comment]
> adding PUID / PGID as environment parameters in docker-compose.yaml works for me

@MrOnak I don't see `PUID` mentioned anywhere in the code, so it is very surprising to see that it works for some people. Is the owner of your `homepage` files and processes really `1002:100`?

What does
```sh
docker exec homepage ps aux
```
return for you?

**Comment by MrOnak** [comment]
@schklom sorry for the late reply and a happy new year to everyone. 
1002:100 matches the user / group id on my host

`docker exec homepage ps aux` does yield this:

```
$ docker exec homepage ps aux
PID   USER     TIME  COMMAND
    1 root      5:47 node server.js
304478 root      0:00 ps aux
```

**Comment by schklom** [comment]
@MrOnak No worries, and happy new year!
Well, the container's main process runs as `root` and not as user 1002, so PUID and PGID did not work there, and likely did nothing.

**Comment by MrOnak** [comment]
@schklom This is odd. I've spun up a new environment that I needed for testing anyway and updated an older version of homepage there. When I remove the PUID/PGID env parameters from the compose file the yaml files used by the container still remain with user/group settings of my non-privileged user. 

So yes, the file ownership isn't affected by the env paramters as you say. 

What I didn't test though is spinning up a new container without existing volume / yaml files. so not sure what default user/group settings new files would have. But as long as homepage doesn't change them on container startup it seems almost irrelevant.

Lastly: I've now removed the PUID/PGID parameters from my "actual" homepage installation (the one where I initially had all .yaml files as root:root) and there too all files keep their user/group.

False alarm I guess. Odd.

**Comment by schklom** [comment]
> the yaml files used by the container still remain with user/group settings of my non-privileged user

If they are only read, there is no reason the user/group would change.
For example, as an unprivileged user, despite `etc/os-release` being owned by root:root, I can read that file with `cat /etc/os-release` and its user:group does not change.

In the meantime, you can specify `read_only` to that container (`read_only: true` https://docs.docker.com/compose/compose-file/compose-file-v3/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir with docker-compose) to add some security.

**Comment by KillahB33** [comment]
This should be mark as closed, @shamoon just merged this feature into main.

**Comment by shamoon** [comment]
Closed by #644
